File: Backend/app/services/weather.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.getenv("OPENWEATHER_API_KEY")

def get_weather(city: str):
    if not API_KEY:
        raise Exception("OPENWEATHER_API_KEY not set")

    url = "https://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city,
        "appid": API_KEY,
        "units": "metric"
    }

    response = requests.get(url, params=params)
    data = response.json()

    if response.status_code != 200:
        raise Exception(data.get("message", "Weather API error"))

    return {
        "city": city,
        "temperature": data["main"]["temp"],
        "condition": data["weather"][0]["description"],
        "humidity": data["main"]["humidity"]
    }

File: Frontend/src/api/weather.js
// src/api/weather.js
import api from "./api";

export const getTripWeather = async (city, startDate, endDate) => {
  const res = await api.get("/weather/forecast", {
    params: {
      city,
      start_date: startDate,
      end_date: endDate,
    },
  });
  return res.data;
};
